### PR TITLE
Add several unit tests for NetworkEntity in Multiplayer Gem

### PR DESCRIPTION
**Pull request by AMZN-puvvadar**
Signed-off-by: puvvadar <puvvadar@amazon.com>

## What does this PR do?

This change introduces a significant number of unit tests to the Multiplayer Gem. These units cover various pieces of basic functionality across the NetworkEntity classes in addition to shoring up a few other areas.

## How was this PR tested?

Ran Multiplayer.Tests and verified completeness with OpenCppCoverage.

### CODE REVIEW

**Review by AMZN-Olex — COMMENTED** [review]
I would love to see the before and after screenshots of OpenCppCoverage results for this.

File: Gems/Multiplayer/Code/Tests/NetworkEntityTests.cpp [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+        message = NetworkEntityUpdateMessage(NetEntityRole::Authority, m_root->m_netId);
+        EXPECT_EQ(message.GetNetworkRole(), NetEntityRole::Authority);
+        message = NetworkEntityUpdateMessage(NetEntityRole::Authority, m_root->m_netId, prefabId);
+        AzNetworking::PacketEncodingBuffer buffer;
+        EXPECT_NE(message.GetPrefabEntityId().m_prefabName.GetCStr(), "");
+        message.SetData(buffer);
+        NetworkEntityUpdateMessage message2(AZStd::move(message));
+        EXPECT_EQ(message, message2);
+        EXPECT_TRUE(m_root->m_replicator->GetPropertyPublisher()->PrepareSerialization());
+        const NetworkEntityUpdateMessage constMessage = m_root->m_replicator->GenerateUpdatePacket();
```
Comment (AMZN-Olex) [inline Gems/Multiplayer/Code/Tests/NetworkEntityTests.cpp:350]: Can this be used to trigger updates in network properties?

**Review by AMZN-Olex — APPROVED** [review]

File: Gems/Multiplayer/Code/Tests/NetworkEntityTests.cpp (hunk 1 above)
Comment (AMZN-puvvadar) [reply Gems/Multiplayer/Code/Tests/NetworkEntityTests.cpp:350]: Most actual network communication is pretty hard to do here unfortunately. We can basically exercise basic functionality and steps that prepare/serialize/deserialize.  In general, I think that's a pretty good goal for network based unit tests.

**Review by AMZN-Gene — APPROVED** [review]

File: Gems/Multiplayer/Code/Source/MultiplayerStats.cpp [hunk 2]
```diff
@@ -60,10 +68,18 @@ namespace Multiplayer
...
+        {
+            m_componentStats[netComponentIndex].m_propertyUpdatesRecv[propertyIndex].m_totalCalls++;
+            m_componentStats[netComponentIndex].m_propertyUpdatesRecv[propertyIndex].m_totalBytes += totalBytes;
+            m_componentStats[netComponentIndex].m_propertyUpdatesRecv[propertyIndex].m_callHistory[m_recordMetricIndex]++;
+            m_componentStats[netComponentIndex].m_propertyUpdatesRecv[propertyIndex].m_byteHistory[m_recordMetricIndex] += totalBytes;
+        }
+        else
+        {
+            AZ_Warning("MultiplayerStats", false,
+                "Component ID %u has fewer than %u propertyIndex. Mismatch by caller suspected.", netComponentIndex, propertyIndex);
```
[the remark this answers is not in the document]
Comment (AMZN-puvvadar) [reply Gems/Multiplayer/Code/Source/MultiplayerStats.cpp:81]: Yup, good call.

**Pull request merged**